General Purpose Vis Bar Chart and Various Improvements

dvzacharycutler opened the pull request:
**Features**

Removed Plotly toolbar, added the functionality for switching between brushes as buttons. Also added opacity slider here. Not sure how this should work with non scatterplots, and I just left it for now. Hiding it makes some sense to me, as selection/opacity changes dont make a ton of sense right now in the other plots, but could be useful at some point. 

![Screenshot from 2021-09-20 17-31-45](https://user-images.githubusercontent.com/87644761/134029841-beec0fbe-8f0d-43b1-afa9-b334dca943b7.png)

Sidepanel improvements, including 

- Making Numerical and Categorical dropdowns multi selectable without closing
- Always show advanced options
- Resort advanced options, removing those that dont make sense for different visualizations
- Removed the many buttons to switch between vis, made it just a dropdown
- 
![Screenshot from 2021-09-20 17-35-05](https://user-images.githubusercontent.com/87644761/134030363-e7902725-e141-40b4-9d4e-17da609b2c04.png)

Label improvements for scatter plots. Similar label issues still need to be handled in other plots in a future ticket. 

![Screenshot from 2021-09-20 17-35-56](https://user-images.githubusercontent.com/87644761/134030446-e6355500-7d55-4b91-b1f5-2c1f9175f752.png)

Scatterplot coloring by numerical attribute. Added both sequential and divergent coloring. This works okay, but the divergent colors need to be changed to have a less white color in the middle. Additionally, usability wise it is not clear how to use this feature, as you first need to select a numerical column to activate the option to switch to divergent/sequential via the sidebar. 

![Screenshot from 2021-09-20 17-38-41](https://user-images.githubusercontent.com/87644761/134031128-a9d7ce73-32c6-4fe3-ac26-71efb88e21ef.png)


Bar Chart is functional, though with some usability problems. It is not clear that the numerical columns do nothing, or that the categorical columns after the first also do nothing. Also need to shorten labels, similar to scatterplot. Grouping/small multiples works. 

![Screenshot from 2021-09-20 17-40-48](https://user-images.githubusercontent.com/87644761/134031402-04be489d-6d85-4a73-bc31-ebe7ed1ddf46.png)

dg-datavisyn reviewed (changes requested):
Already looks great 👍 


Some issues I noticed:

## Scatter Chart

- [x] Scatterplot color by numeric value looks strange. I colored by `pvalue` which is also the y-axis. So I expected a smooth gradient along the y axis
![image](https://user-images.githubusercontent.com/45822063/134033139-0a964f13-86ea-4bc1-847c-dc643678b574.png)
- [x] Please add tooltips to the selection type changes (brush, lasso) 
![image](https://user-images.githubusercontent.com/45822063/134033228-142d334f-4c26-4b3e-bdba-5ff1d936d2ff.png)
- [x] Could we use a bounding box icon for brush selection and a lasso icon for lasso selection?
- [x] Selection is really slow (maybe an extra ticket)
- [x] Opacity change is slow, maybe add a debounce to the slider
- [x] Could we add little color ranges to the `Divergent` and `Sequential` buttons so that is clear to the user what is meant here, like this (but horizontal): 
![image](https://user-images.githubusercontent.com/45822063/134032968-e13a5712-3f8c-4241-804f-669e4fd46e2a.png)
- [ ] Please add a tooltip to the `Divergent` and `Sequential` buttons as well as to the `Filter In | Filter Out | Clear`
- [ ] There is no legend for the color
- [x] :bug: `Filter In | Filter Out | Clear` buttons aren't working (Related question: Does this make sense if the component is used without lineup?)

## Bar Chart

- [x] When I switch to bar chart it takes an integer attribute on the x-axis but it is not listed: 
![image](https://user-images.githubusercontent.com/45822063/134034029-8c749b09-9c4e-497c-80ce-3f220b039cf6.png)
- [ ] The numerical columns are not used in the bar chart
- [x] When I switch from vertical to horizontal the axes labels are not switched: 
![image](https://user-images.githubusercontent.com/45822063/134034373-aa5ca769-b3e4-425a-a291-722a185d8bca.png)

dvzacharycutler commented:
Fixed most of the mentioned problems, which were primarily bug fixes. The only new stuff that wasn't a bug fix is:

1. Selection is much faster. Simplified the way I stored selection to move an n^2 selection time to n. Remaining improvements could be made to lineup itself to increase selection speed. 
2. Added color scale beneath the sequential/divergent buttons
![Screenshot from 2021-09-22 15-43-06](https://user-images.githubusercontent.com/87644761/134355039-cbbe8427-9af4-420d-bbc7-3abc6df0de3c.png)

thinkh commented:
We agreed that functionality-wise the PR is done and code-wise it will be changed/improved with the refactor in the next PR. Hence, I'll continue and merge this PR.
